### PR TITLE
migration: Update migration_poweroff_vm case

### PR DESCRIPTION
**Pull request by cliping**
Add a new attribute in test matrix part:
  New attribute
    NBD
    non-NBD

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

 (1/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.non_nbd.with_precopy.p2p: PASS (248.11 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.non_nbd.with_precopy.non_p2p: PASS (247.86 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.non_nbd.with_postcopy.p2p: PASS (325.18 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.non_nbd.with_postcopy.non_p2p: PASS (366.94 s)

 (1/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.nbd.with_precopy.p2p: PASS (207.71 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.nbd.with_precopy.non_p2p: PASS (208.32 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.nbd.with_postcopy.p2p: PASS (208.80 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.nbd.with_postcopy.non_p2p: PASS (208.76 s)

**Review by Yingshun — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
lgtm

File: libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py [hunk 1]
```diff
@@ -67,10 +82,12 @@ def verify_poweroff_vm():
         locals() else migration_obj.setup_connection
     verify_test = eval("verify_%s" % test_case) if "verify_%s" % test_case in \
         locals() else migration_obj.verify_default
+    cleanup_test = eval("cleanup_%s" % test_case) if "cleanup_%s" % test_case in \
```
Comment (hholoubk) [inline libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py:85]: If there is not other testcases .. the definition here will not be necessary ... as it is always hard to me quickly realize what does this code mean, and why we need to eval it in this way (I don't like eval anyway, as I feel it as not secure way to solve things. 

I might be wrong, and I would be glad to understand reasons for thins

File: libvirt/tests/cfg/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.cfg [hunk 2]
```diff
@@ -20,27 +15,43 @@
...
         - with_precopy:
-            virsh_migrate_src_state = "shut off"
         - with_postcopy:
-            postcopy_options = '--postcopy --timeout 10 --timeout-postcopy --postcopy-bandwidth 5'
-            virsh_migrate_src_state = "paused"
-            poweroff_vm_dest = "yes"
+            postcopy_options = '--postcopy --timeout 5 --timeout-postcopy --postcopy-bandwidth 5'
+    variants:
+        - nbd:
+            virsh_migrate_extra = "--bandwidth 5 --copy-storage-all --migrate-disks vda"
```
Comment (hholoubk) [inline libvirt/tests/cfg/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.cfg:43]: I've checked the original test steps in XXX-294690 definition and there is one more parameter mentioned in the NBD part ... 

--copy-storage-inc

are we going to do also this migration?

File: libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py [hunk 3]
```diff
@@ -67,10 +82,12 @@ def verify_poweroff_vm():
...
+        locals() else migration_obj.cleanup_connection
 
     try:
         setup_test()
         migration_obj.run_migration()
         verify_test()
     finally:
-        migration_obj.cleanup_connection()
+        cleanup_test()
```
Comment (hholoubk) [inline libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py:93]: I know that this was already been in the original case, but 
... If the only reason is that it is done same way in other scripts, here we are using only one method to setup/verify/cleanup. 

We should have here direct names of methods and not define them via eval.

**Comment by Yingshun** [comment]
@hholoubk  This pr has been merged. If you have any questions, you can contact lcheng or submit another pr. Thx!

**Pull request merged**